executor/insert_common.go

```diff
@@ -941,6 +952,20 @@ func (e *InsertValues) batchCheckAndInsert(ctx context.Context, rows [][]types.D
 		return err
 	}
 
+	e.snapshot = txn.GetSnapshot()
+	if e.runtimeStats != nil {
+		snapshotStats := &tikv.SnapshotRuntimeStats{}
+		e.stats = &RuntimeStatsWithRPCStats{
+			BasicRuntimeStats:    e.runtimeStats,
+			SnapshotRuntimeStats: snapshotStats,
+		}
+		e.snapshot.SetOption(kv.CollectRuntimeStats, snapshotStats)
```

Comment:
Also need add `defer e.snapshot.DelOption(kv.CollectRuntimeStats)` before this function return.

Reply:
This line have already been added before the final return in the function, please check out the latest commit.